pkg/sdk/model/output/http.go

```diff
@@ -0,0 +1,147 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package output
+
+import (
+	"github.com/banzaicloud/operator-tools/pkg/secret"
+
+	"github.com/banzaicloud/logging-operator/pkg/sdk/model/types"
+)
+
+// +name:"Http"
+// +weight:"200"
+type _hugoHTTP interface{}
+
+// +docName:"Http plugin for Fluentd"
+// Sends logs to HTTP/HTTPS endpoints.
+// More info at https://docs.fluentd.org/output/http.
+//
+// #### Example output configurations
+// ```
+// spec:
+//   http:
+//     endpoint: http://logserver.com:9000/api
+//     buffer:
+//       tags: "[]"
+//       flush_interval: 10s
+// ```
+type _docHTTP interface{}
+
+// +name:"Http"
+// +url:"https://docs.fluentd.org/output/http"
+// +version:"more info"
+// +description:"Sends logs to HTTP/HTTPS endpoints."
+// +status:"GA"
+type _metaHTTP interface{}
+
+// +kubebuilder:object:generate=true
+// +docName:"Output Config"
+type HTTPOutputConfig struct {
+	// Endpoint for HTTP request.
+	Endpoint string `json:"endpoint"`
+	// Method for HTTP request. [post, put] (default: post)
+	HTTPMethod string `json:"http_method,omitempty"`
+	// Proxy for HTTP request.
+	Proxy string `json:"proxy,omitempty"`
+	// Content-Type for HTTP request.
+	ContentType string `json:"content_type,omitempty"`
+	// +docLink:"Format,../format/"
+	Format *Format `json:"format,omitempty"`
+	// Additional headers for HTTP request.
+	Headers map[string]string `json:"headers,omitempty"`
+	// Connection open timeout in seconds.
+	OpenTimeout int `json:"open_timeout,omitempty"`
+	// Read timeout in seconds.
+	ReadTimeout int `json:"read_timeout,omitempty"`
+	// TLS timeout in seconds.
+	SSLTimeout int `json:"ssl_timeout,omitempty"`
+	// The default version of TLS transport. [TLSv1_1, TLSv1_2] (default: TLSv1_2)
+	TlsVersion string `json:"tls_version,omitempty"`
+	// The cipher configuration of TLS transport. (default: ALL:!aNULL:!eNULL:!SSLv2)
+	TlsCiphers string `json:"tls_ciphers,omitempty"`
+	// The CA certificate path for TLS.
+	TlsCACertPath *secret.Secret `json:"tls_ca_cert_path,omitempty"`
+	// The client certificate path for TLS.
+	TlsClientCertPath *secret.Secret `json:"tls_client_cert_path,omitempty"`
+	// The client private key path for TLS.
+	TlsPrivateKeyPath *secret.Secret `json:"tls_private_key_path,omitempty"`
+	// The client private key passphrase for TLS.
+	TlsPrivateKeyPassphrase *secret.Secret `json:"tls_private_key_passphrase,omitempty"`
+	// The verify mode of TLS. [peer, none] (default: peer)
+	TlsVerifyMode string `json:"tls_verify_mode,omitempty"`
+	// Raise UnrecoverableError when the response code is non success, 1xx/3xx/4xx/5xx. If false, the plugin logs error message instead of raising UnrecoverableError. (default: true)
+	ErrorResponseAsUnrecoverable bool `json:"error_response_as_unrecoverable,omitempty"`
```

Comment:
```suggestion
	ErrorResponseAsUnrecoverable *bool `json:"error_response_as_unrecoverable,omitempty"`
```
so that a false value won't be cleared by omitempty